history.go

```diff
@@ -7,37 +7,60 @@ import (
 )
 
 var (
-	ErrBadEpoch    = errors.New("[merkletree] Next epoch's STR has bad epoch")
-	ErrSTRNotFound = errors.New("[merkletree] STR not found")
+	ErrBadEpoch           = errors.New("[merkletree] Next epoch's STR has bad epoch")
+	ErrSTRNotFound        = errors.New("[merkletree] STR not found")
+	ErrMalformedHashChain = errors.New("[merkletree] Malformed hash chain")
 )
 
+type Epoch int64
+
 type History struct {
-	str           *SignedTreeRoot
-	epochInterval int64
+	currentSTR    *SignedTreeRoot
+	hashChain     map[Epoch]*SignedTreeRoot
+	epochs        []Epoch // slice of epochs in hashChain
```

Comment:
I use this array to delete older values stored in the map, since the order of the keys in golang map is randomized. (https://github.com/coniks-sys/libmerkleprefixtree-go/blob/496b4a6c345678de2a7fcec3f35b0e91c4fa6da9/history.go#L49)